db/db_impl.cc

```diff
@@ -4759,18 +4759,18 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
       if (!cfd->mem()->IsEmpty()) {
         // We only consider active mem table, hoping immutable memtable is
         // already in the process of flushing.
-        size_t cfd_size = cfd->mem()->ApproximateMemoryUsage();
-        if (largest_cfd == nullptr || cfd_size > largest_cfd_size) {
-          largest_cfd = cfd;
-          largest_cfd_size = cfd_size;
+        uint64_t seq = cfd->mem()->GetLatestSeqWhenCreate();
```

Comment:
Do you see any advantages of using GetLatestSeqWhenCreate() instead of GetFirstSequenceNumber() here? I can't think of any. Some advantages of GetFirstSequenceNumber():
 * Has a property that's important for logdevice: suppose you wrote a key-value pair A, then wrote db_write_buffer_size bytes of data to the DB (any CFs). Write A is (almost) guaranteed to be have been flushed by now. Imo this also makes it easy to reason about.
 * WAL never grows above 2*db_write_buffer_size. (Of course there's max_total_wal_size, but in logdevice we're going to stop using WAL and have our own external WAL-like thing that we don't want to grow too much; max_total_wal_size won't help with that.)
 * Less code.

A (fairly realistic with our workload) scenario in which GetLatestSeqWhenCreate() fails the first two advantages:
 * db_write_buffer_size = 1 GB. There are 1000 CFs. No one writes to them atm.
 * A new CF is created, a bunch of stuff is written to it (some flushes happen), then writes to it stop. A small amount of data is still left in its memtable.
 * 900 MB are written to CF 1, then 900 MB are written to CF 2 and so on up to CF 1000. During this whole time the memtable in CF 1001 stays unflushed. WAL grows to 900 GB.